### ISSUE TITLE
OmniAuth v2.0.0rc1

### ISSUE THREAD

**Issue by BobbyMcWho** [toyokazu/omniauth-shibboleth#23]
Hello, maintainer of OmniAuth here. 

I just wanted to make the maintainers of this gem aware of the discussion that I have opened regarding v2.0.0 of OmniAuth. I invite you to join in and voice any concerns you may have here: https://github.com/omniauth/omniauth/discussions/1017

**Comment by DaAwesomeP** [comment]
OmniAuth 2 has been released. @toyokazu are there any known incompatibilities? Is it safe to migrate to OmniAuth 2? The repo description specifically says 1.x.

**Comment by rrrrrrmb** [comment]
@BobbyMcWho Hey there. I'm working on getting omniauth 2.0.x into Debian. We have a few build failures for packages that depend on omniauth with 2.0.x version.

omniauth-shibboleth failed to build with the following errors:
```>
Failure/Error: expect(last_response.status).to eq(302)
 
        expected: 302
             got: 200
 
        (compared using ==)
      # ./spec/omniauth/strategies/shibboleth_spec.rb:42:in `block (3 levels) in <top (required)>'
 
 Finished in 0.06473 seconds (files took 0.14061 seconds to load)
 16 examples, 1 failure
 
 Failed examples:
 
 rspec ./spec/omniauth/strategies/shibboleth_spec.rb:41 # OmniAuth::Strategies::Shibboleth request phase is expected to redirect o callback_url
 
/usr/bin/ruby2.7 -I/usr/share/rubygems-integration/all/gems/rspec-support-3.9.3/lib:/usr/share/rubygems-integration/all/gems/rspec-core-3.9.2/lib /usr/share/rubygems-integration/all/gems/rspec-core-3.9.2/exe/rspec --pattern ./spec/\*\*/\*_spec.rb --format documentation failed
 ERROR: Test "ruby2.7" failed: 
```

From reading the release notes of omniauth 2 I got to know that the default method for request_phase is now POST only. So I think I should rewrite the [test](https://github.com/toyokazu/omniauth-shibboleth/blob/a3de67b1b3a29ce0e004898c0fb6dd6a6afe70ab/spec/omniauth/strategies/shibboleth_spec.rb#L41) for the callback_url

```
  context 'request phase' do
    before do
      get '/auth/shibboleth'
    end

    it 'is expected not to redirect to callback_url' do
      expect(last_response.status).not_to eq(302)

    end
```

Since the upstream has'nt been responsive/active for a long time, I wanted to include this as a patch for the debian package. I needed your thoughts on this change. Since I'm not that much familiar with ruby :)

**Comment by DaAwesomeP** [comment]
@mdb571, I don't think you should patch anything in a Debian package that isn't a Debian-specific bug. The issue is not with how you are planning to package the repo but rather with that this repo has not been updated to properly support the new omniauth. Package-level patches are meant to introduce Debian-specific modifications or patch issues that arise with installation on that specific system.

You would be much better off forking this repository and calling it omniauth-shibboleth2 or something like that and fixing all of the bugs properly without Debian-specific patches.

**Comment by DaAwesomeP** [comment]
My apologies, I meant to mention @mdb571 not the original issue-opener. I have fixed it now.

**Comment by rrrrrrmb** [comment]
@DaAwesomeP Debian has bug [report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=999726) about a test regression with omniauth 2. Currently I don't have the skills to maintain a fork of this package. So I thought of adding a patch to the debian package that fixes this test., SInce the original author hasn't made the necessary updates to support omniauth 2

If you could help. Can you suggest the appropriate changes that has to be made inorder to fix this regression ? I'll add that change as patch so we can have a smooth transition for Omniauth 2 into debian

**Comment by DaAwesomeP** [comment]
Just a note for the interested, I have moved to use omniauth-saml to directly authenticate with the IdP server without mod_shib/shibd. Doesn't work for all situations, but omniauth-saml is much more maintained.

**Comment by alansill** [comment]
See related issue #25
